src/compiler/parser.ts

```diff
@@ -2472,6 +2482,9 @@ namespace ts {
                     return parseTupleType();
                 case SyntaxKind.OpenParenToken:
                     return parseParenthesizedType();
+                case SyntaxKind.PartialKeyword:
+                    const partialNode = tryParse(parsePartialType);

```

Comment:
Maybe we should restrict the grammar to `partial TypeReference` ?. It doesn't make any sense to allow `partial ObjectTypeLiteral`.